lib/xpm_ruby/connection.rb

```diff
@@ -12,12 +12,48 @@ def initialize(access_token:, xero_tenant_id:)
 
     def get(endpoint:, params: nil)
       faraday_connection = Faraday.new(url)
-      faraday_connection.get(endpoint, params, headers)
+      response = faraday_connection.get(endpoint, params, headers)
+
+      case response.status
+      when 401
+        hash = Ox.load(response.body, mode: :hash_no_attrs, symbolize_keys: false)
+
+        raise Unauthorized.new(hash["html"]["head"]["title"])
+      when 200
+        xml = Ox.load(response.body, mode: :hash_no_attrs, symbolize_keys: false)
+
+        case xml["Response"]["Status"]
+        when "OK"
+          xml["Response"]
+        when "ERROR"
+          raise Error.new(response["ErrorDescription"])
+        end
+      else
+        raise Error.new(response.status)
+      end
     end
 
     def post(endpoint:, data:)
       faraday_connection = Faraday.new(url)
-      faraday_connection.post(endpoint, data, headers)
+      response = faraday_connection.post(endpoint, data, headers)
+
+      case response.status
+      when 401
+        hash = Ox.load(response.body, mode: :hash_no_attrs, symbolize_keys: false)
+
+        raise Unauthorized.new(hash["html"]["head"]["title"])
+      when 200
+        xml = Ox.load(response.body, mode: :hash_no_attrs, symbolize_keys: false)
+
+        case xml["Response"]["Status"]
+        when "OK"
+          xml["Response"]
+        when "ERROR"
+          raise Error.new(response["ErrorDescription"])
+        end
+      else
+        raise Error.new(response.status)
+      end
     end
```

Comment:
As I mentioned in the description we can clean this up later at least it's in once place now.